src/plugins/vis_augmenter/public/view_events_flyout/components/view_events_flyout.tsx

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useEffect } from 'react';
+import {
+  EuiFlyoutBody,
+  EuiFlyoutHeader,
+  EuiFlyout,
+  EuiTitle,
+  EuiFlexGroup,
+  EuiFlexItem,
+} from '@elastic/eui';
+import './styles.scss';
+import { VisualizeEmbeddable } from '../../../../visualizations/public';
+import { TimeRange } from '../../../../data/common';
+import { BaseVisItem } from './base_vis_item';
+import { DateRangeItem } from './date_range_item';
+import { LoadingFlyoutBody } from './loading_flyout_body';
+import { ErrorFlyoutBody } from './error_flyout_body';
+import { EventsPanel } from './events_panel';
+import { TimelinePanel } from './timeline_panel';
+import { fetchVisEmbeddable, createEventEmbeddables, createTimelineEmbeddable } from './utils';
+import { EventVisEmbeddablesMap } from './types';
+
+interface Props {
+  onClose: () => void;
+  savedObjectId: string;
+}
+
+export const DATE_RANGE_FORMAT = 'MM/DD/YYYY HH:mm';
+
+export function ViewEventsFlyout(props: Props) {
+  const [visEmbeddable, setVisEmbeddable] = useState<VisualizeEmbeddable | undefined>(undefined);
+  // This map persists a plugin resource type -> a list of vis embeddables
+  // for each VisLayer of that type
+  const [eventVisEmbeddablesMap, setEventVisEmbeddablesMap] = useState<
+    EventVisEmbeddablesMap | undefined
+  >(undefined);
+  const [timelineVisEmbeddable, setTimelineVisEmbeddable] = useState<
+    VisualizeEmbeddable | undefined
+  >(undefined);
+  const [timeRange, setTimeRange] = useState<TimeRange | undefined>(undefined);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [errorMessage, setErrorMessage] = useState<string | undefined>(undefined);
+
+  function reload() {
+    visEmbeddable?.reload();
+    eventVisEmbeddablesMap?.forEach((embeddableItems) => {
+      embeddableItems.forEach((embeddableItem) => {
+        embeddableItem.embeddable.reload();
+      });
+    });
+  }
+
+  useEffect(() => {
+    fetchVisEmbeddable(props.savedObjectId, setTimeRange, setVisEmbeddable, setErrorMessage);
+    /* eslint-disable */
+  }, [props.savedObjectId]);
+
+  useEffect(() => {
+    if (visEmbeddable?.visLayers) {
+      createEventEmbeddables(
+        props.savedObjectId,
+        visEmbeddable,
+        setEventVisEmbeddablesMap,
+        setErrorMessage
+      );
+      createTimelineEmbeddable(
+        props.savedObjectId,
+        visEmbeddable,
+        setTimelineVisEmbeddable,
+        setErrorMessage
+      );
+    }
+  }, [visEmbeddable?.visLayers]);
+
+  useEffect(() => {
+    if (
+      visEmbeddable !== undefined &&
+      eventVisEmbeddablesMap !== undefined &&
+      timeRange !== undefined &&
+      timelineVisEmbeddable !== undefined
+    ) {
+      setIsLoading(false);
+    }
```

Comment:
I haven't ran into that case yet - typically I've relied on the vis embeddables to handle those scenarios - for example, if there is some error within the embeddable it will just render an error embeddable with the error message. I've reproduced this a bunch of times. This is why i only do undefined checks here.

I'll update after more plugin integration testing and see if we can trigger an infinite loading case.